lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -783,6 +784,54 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
   @override
   bool get alwaysUse24HourFormat => configuration.alwaysUse24HourFormat;
 
+  /// Updates [textScaleFactor] and invokes [onTextScaleFactorChanged]
+  /// callback if [textScaleFactor] changed.
+  void _updateTextScaleFactor(double value) {
+    if (configuration.textScaleFactor != value) {
+      _configuration = configuration.copyWith(textScaleFactor: value);
+      invokeOnPlatformConfigurationChanged();
+      invokeOnTextScaleFactorChanged();
+    }
+  }
+
+  /// Watches for font-size changes in the browser's <html> element to
+  /// recalculate [textScaleFactor].
+  ///
+  /// Updates [textScaleFactor] with the new value.
+  html.MutationObserver? _fontSizeObserver;
+
+  /// Set the callback function for updating [textScaleFactor] based on
+  /// font-size changes in the browser's <html> element.
+  void _observeFontSize() {
+    const String styleAttribute = 'style';
+
+    _fontSizeObserver = html.MutationObserver(
+        (List<dynamic> mutations, html.MutationObserver _) {
+      for (final html.MutationRecord record
+          in mutations.cast<html.MutationRecord>()) {
```

Comment:
Instead of casting the entire list, thus unnecessarily creating a new list, it might be better to cast each item individually inside the body of the for loop. @yjbanov any thoughts here?

Reply:
I did this based on the precedent of [scene_builder_test.dart](https://github.com/flutter/engine/blob/main/lib/web_ui/test/engine/surface/scene_builder_test.dart#L503). Should we also change it there?

Reply:
We don't care about performance of test code. This code will run in production though, so it's worth tightening it up.